test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -712,4 +728,112 @@ private static List<string> ParseDebControlDependencies(string contents)
 
         return results;
     }
+
+    private void ValidatePackageList(PackageType packageType)
+    {
+        string extension = packageType == PackageType.Rpm ? "*.rpm" : "*.deb";
+        List<string> expectedPatterns = GetExpectedPackagePatterns(packageType).OrderBy(p => p).ToList();
+
+        // Find all packages of the specified type and normalize by removing version numbers
+        List<string> normalizedActual = Directory.GetFiles(Config.AssetsDirectory, extension, SearchOption.AllDirectories)
+            .Select(path => RemoveVersionFromPackageNameRegex.Replace(Path.GetFileName(path), "*"))
+            .Distinct()
+            .OrderBy(name => name)
+            .ToList();
+
+        // Compare the lists
+        var missing = expectedPatterns.Except(normalizedActual).ToList();
+        var extra = normalizedActual.Except(expectedPatterns).ToList();
+
+        if (missing.Any() || extra.Any())
+        {
+            _outputHelper.WriteLine($"Expected {packageType} packages:");
+            foreach (string pattern in expectedPatterns)
+            {
+                _outputHelper.WriteLine($"  {pattern}");
+            }
+
+            _outputHelper.WriteLine($"\nActual {packageType} packages (normalized):");
+            foreach (string package in normalizedActual)
+            {
+                _outputHelper.WriteLine($"  {package}");
+            }
+
+            var errorMessage = new StringBuilder();
+            errorMessage.AppendLine($"Package list validation failed for {packageType}:");
+
+            if (missing.Any())
+            {
+                errorMessage.AppendLine("Missing packages:");
+                foreach (string pkg in missing)
+                {
+                    errorMessage.AppendLine($"  - {pkg}");
+                }
+            }
+
+            if (extra.Any())
+            {
+                errorMessage.AppendLine("Unexpected packages:");
+                foreach (string pkg in extra)
+                {
+                    errorMessage.AppendLine($"  + {pkg}");
+                }
+            }
+
+            Assert.Fail(errorMessage.ToString());
+        }
```

Comment:
With `FluentAssertions`, this can be simplified to:
```suggestion
        normalizedActual.Should().BeEquivalentTo(expectedPatterns);
```